pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -46,10 +49,16 @@ func NewPgxIngestor(conn pgxconn.PgxConn, cache cache.MetricCache, sCache cache.
 	if err != nil {
 		return nil, err
 	}
+
+	batcherConfg := trace.BatcherConfig{
+		MaxBatchSize: cfg.TracingMaxBatchSize,
+		BatchTimeout: cfg.TracingBatchTimeout,
+		Writers:      cfg.NumCopiers,
+	}
 	return &DBIngestor{
 		sCache:     sCache,
 		dispatcher: dispatcher,
-		tWriter:    trace.NewWriter(conn),
+		tWriter:    trace.NewDispatcher(trace.NewWriter(conn), cfg.AsyncAcksTraces, batcherConfg),
```

Comment:
I think `AsyncAcksTraces` should be part of batcherConfig, rather than sending it as separate param.

Is it because of 1 each for reading and writing?

Reply:
It is not part of Batcher configuration since Batcher does not care if it's async or not. It will just batch and write batches. The dispatcher is actually affaeted by async flag - so it will either return immediatelly or wait until request is persisted.